### PR TITLE
Imsrawdataoverview

### PR DESCRIPTION
**Pull request by SteffenHeu**
still work in progress
grafik

### CODE REVIEW

**Comment by SteffenHeu** [comment]
grafik

I know its an incredibly dirty way to adjust the plots, but it's the only thing i could think of. What do you think?

**Comment by SteffenHeu** [comment]
grafik

There's one small thing i want to fix tomorrow and then i'd say it's done.

**Comment by SteffenHeu** [comment]
From my side it's ready now.

File: src/main/java/io/github/mzmine/modules/visualization/rawdataoverviewims/Threads.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import io.github.mzmine.util.color.SimpleColorPalette;
+import java.awt.Color;
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javafx.application.Platform;
+import javax.annotation.Nonnull;
+
+class Threads {
```
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/modules/visualization/rawdataoverviewims/Threads.java:42]: What is the Threads class for? The individual classes could be defined in their own individual files, no?

File: src/main/java/io/github/mzmine/modules/visualization/rawdataoverviewims/IMSRawDataOverviewTask.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  public double getFinishedPercentage() {
+    return 0;
+  }
+
+  @Override
+  public void run() {
+    setStatus(TaskStatus.PROCESSING);
+    MZmineTab tab = new IMSRawDataOverviewTab(parameterSet);
+    tab.onRawDataFileSelectionChanged(Set.of(file));
+    MZmineCore.getDesktop().addTab(tab);
```
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/modules/visualization/rawdataoverviewims/IMSRawDataOverviewTask.java:56]: This design is a bit strange - it seems like the only purpose of this Task is to add the new tab to the desktop. Tasks should be for data processing, so the tab can just be added immediately in the runModule() method

File: src/main/java/io/github/mzmine/modules/visualization/rawdataoverviewims/Threads.java (hunk 1 above)
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/visualization/rawdataoverviewims/Threads.java:42]: Yes they could. I thought it was unnecessary to make three classes for such rudimentary tasks and they are all somehow related. Should i split it up any way?

Comment (tomas-pluskal) [reply src/main/java/io/github/mzmine/modules/visualization/rawdataoverviewims/Threads.java:42]: I think short and simple classes are better for readability. You can move them to a separate subpackage.

**Comment by tomas-pluskal** [comment]
Looks good, thanks!

**Pull request merged**